src/osd/modules/file/winfile.cpp

```diff
@@ -184,14 +184,7 @@ osd_file::error osd_file::open(std::string const &orig_path, UINT32 openflags, p
 		return win_open_ptty(path, openflags, file, filesize);
 
 	// convert path to TCHAR
-	TCHAR *t_path = tstring_from_utf8(path.c_str());
-	osd_disposer<TCHAR> t_path_disposer(t_path);
-	if (!t_path)
-		return error::OUT_OF_MEMORY;
-
-	// convert the path into something Windows compatible
-	for (TCHAR *src = t_path; *src != 0; src++)
-		*src = /* ('/' == *src) ? '\\' : */ *src;
+	auto t_path = tstring_from_utf8(path.c_str());
```

Comment:
Can the osd_disposer template be eliminated completely or is it used for stuff other than strings?